src/tokenizer/tokenized_string.rs

```diff
@@ -0,0 +1,188 @@
+use crate::tokenizer::{Token, TokenStream, TokenStreamChain};
+use std::cmp::Ordering;
+
+/// Struct representing tokenized text
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+pub struct TokenizedString {
+    /// Original text
+    pub text: String,
+    /// Tokens derived from the text
+    pub tokens: Vec<Token>,
+}
+
+impl Ord for TokenizedString {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.text.cmp(&other.text)
+    }
+}
+
+impl PartialOrd for TokenizedString {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+/// TokenStream implementation which wraps TokenizedString
+pub struct TokenizedStream {
+    tokenized_string: TokenizedString,
+    current_token: i64,
+}
+
+impl From<&TokenizedString> for TokenizedStream {
+    fn from(s: &TokenizedString) -> TokenizedStream {
+        TokenizedStream {
+            tokenized_string: s.clone(),
```

Comment:
Is the clone() strictly necessary here?

Reply:
Probably it isn't. from should take ownership of s. Is this what you had in mind?